lib/SILGen/SILGenExpr.cpp

```diff
@@ -2293,6 +2293,27 @@ RValue RValueEmitter::visitRebindSelfInConstructorExpr(
   return SGF.emitEmptyTupleRValue(E, C);
 }
 
+static bool isNullableTypeInC(SILModule &M, Type ty) {
+  if (ty->hasReferenceSemantics())
+    return true;
+  
+  auto &C = ty->getASTContext();
+  auto nom = ty->getAnyNominal();
+  if (!nom)
+    return false;
+
+  if (nom == C.getUnsafePointerDecl()
+      || nom == C.getUnsafeMutablePointerDecl()
+      || nom == C.getAutoreleasingUnsafeMutablePointerDecl()
+      || nom == C.getOpaquePointerDecl())
+    return true;
+  
+  if (ty->isEqual(M.Types.getSelectorType()))
+    return true;
+  
+  return false;
+}
```

Comment:
@jrose-apple Updated again to handle `OpaquePointer` and `Selector`. Any other concerns?